### PR TITLE
Enhancements

### PR DESCRIPTION
**Pull request by NikitaKurpas**
I've mode some enhancements to your code:
- regexp support (based on Zackio's fork)
- onNotFound callback (based on Zackio's fork)
- code cleanup
- added more tests
- documented new features

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,10 +25,10 @@ gulp.task('default', function () {
 });
 ```
 
-The contains function accepts either a string or an array of strings (any of
+The contains function accepts either a string, regex or an array of strings (any of
 which, when matched, will cause an error to be thrown).
 
-You can also specify a callback function, in which you can handle the error
+You can also specify an onFound or onNotFound callbacks, in which you can handle the error
```
Comment (callumacrae) [inline README.md:31]: This doesn't make sense, it should either by "onFound or onNotFound callbacks" or "an onFound or onNotFound callback" (preferably the former)

File: README.md [hunk 2]
```diff
@@ -44,12 +44,37 @@ gulp.task('default', function () {
...
 		}));
 });
 ```
 
+Regex
+
+```js
+    .pipe(plugins.contains({
+      search: /([ \t]*\n){4,}/, // Looks for multiple empty lines
+      regex: true,
```
Comment (callumacrae) [inline README.md:68]: Is a `regex` option really needed? Surely you can just detect whether `search` is a string or regex?

File: index.js [hunk 3]
```diff
@@ -4,56 +4,78 @@ var through = require('through2');
...
+            cb(new gutil.PluginError('gulp-contains', 'You did not specify a valid search string'));
+            return;
+        }
+
+        var found = false;
+
+        if (!options.regex) {
+            found = stringContains(file.contents.toString(enc), options.search);
+        } else {
+            var pattern = new RegExp(options.search);
```
Comment (callumacrae) [inline index.js:44]: Could you move this logic into the `stringContains` function? Possibly rename it to `stringMatches`, too.

File: index.js [hunk 4]
```diff
@@ -4,56 +4,78 @@ var through = require('through2');
...
-		}
-	}
+    for (var i = 0; i < search.length; i++) {
+        if (stringContains(str, search[i])) {
+            return search[i];
+        }
+    }
 
-	return false;
+    return false;
```
Comment (callumacrae) [inline index.js:80]: Have you converted the tabs to spaces here? I can't tell what has happened here

**Comment by callumacrae** [comment]
Except for the couple things I mentioned (mostly around the regex logic), this looks good to me :+1:

**Comment by NikitaKurpas** [comment]
Ok, I will do those fixes, close this pull request and create a new one

**Comment by callumacrae** [comment]
Thanks!

**Pull request closed without merging**